qiskit/circuit/instruction.py

```diff
@@ -250,7 +250,7 @@ def inverse(self):
             QiskitError: if the instruction is not composite
                 and an inverse has not been implemented for it.
         """
-        if not self.definition:
+        if self.definition is None:
```

Comment:
why narrow this check? If an empty list comes along shouldn't that still raise this exception?